mesh/pmesh.cpp

```diff
@@ -5568,6 +5568,97 @@ void ParMesh::PrintSharedEntities(const char *fname_prefix) const
    }
 }
 
+void ParMesh::GetGlobalVertexIndices(Array<HYPRE_Int> &gi)
+{
+   H1_FECollection fec(1, Dim); // Order 1, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   MFEM_VERIFY(fespace.GetVSize() == GetNV(),
+               "There should be a bijection between DOFs and vertices");
+
+   gi.SetSize(GetNV());
+
+   Array<int> dofs;
+   bool oneDof = true;
+   for (int i=0; i<GetNV(); ++i)
+   {
+      fespace.GetVertexDofs(i, dofs);
+      oneDof = oneDof && (dofs.Size() == 1);
+      gi[i] = fespace.GetGlobalTDofNumber(dofs[0]);
+   }
+
+   MFEM_VERIFY(oneDof, "Each vertex should have exactly one DOF");
+}
+
+void ParMesh::GetGlobalEdgeIndices(Array<HYPRE_Int> &gi)
+{
+   ND_FECollection fec(1, Dim); // Order 1, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   MFEM_VERIFY(fespace.GetVSize() == GetNEdges(),
+               "There should be a bijection between DOFs and edges");
+
+   gi.SetSize(GetNEdges());
+
+   Array<int> dofs;
+   bool oneDof = true;
+   for (int i=0; i<GetNEdges(); ++i)
+   {
+      fespace.GetEdgeDofs(i, dofs);
+      oneDof = oneDof && (dofs.Size() == 1);
+      const int ldof = (dofs[0] >= 0) ? dofs[0] : -1 - dofs[0];
+      gi[i] = fespace.GetGlobalTDofNumber(ldof);
+   }
+
+   MFEM_VERIFY(oneDof, "Each edge should have exactly one DOF");
+}
+
+void ParMesh::GetGlobalFaceIndices(Array<HYPRE_Int> &gi)
+{
+   MFEM_VERIFY(Dim == 3, "GetGlobalFaceIndices can be called only in 3D");
```

Comment:
In many cases throughout MFEM, calls involving `faces` in 2D fall back to `edges`. Should we have the same here, i.e., calling `GetGlobalEdgeIndices` ?

Reply:
Good point. It would be easy to call `GetGlobalEdgeIndices` in the 2D case (one more line of code). I think it is clearer to force the user to call one function, rather than having the wrong function call the right one. This is a matter of opinion, and I am fine with either choice.

Reply:
I would expect `faces` to work with any dimension. I vote to call the functions for edges and vertices for `dim=2` and `dim=1`.

Reply:
@vladotomov Should the function for edges call the function for vertices in 1D? I find these equivalences in different dimensions more confusing than helpful, but if that is common we can do it here as well for consistency.

Reply:
Yes, see `Mesh::GetNumFaces()`.

Reply:
Thanks @vladotomov. I think all your comments are addressed now.